src/main/java/seedu/address/logic/parser/CustomiseCommandParser.java

```diff
@@ -33,27 +38,30 @@ public CustomiseCommand parse(String args) throws ParseException {
         }
 
         if (arguments[0].equals("order")) {
-            return new CustomiseCommand("order", toAttributeOrder(arguments));
+            return new CustomiseCommand(ORDER, toAttributeOrder(arguments));
+        } else if (arguments[0].equals("hide")) {
+            return new CustomiseCommand(HIDE, toAttributes(arguments));
+        } else if (arguments[0].equals("show")) {
+            return new CustomiseCommand(SHOW, toAttributes(arguments));
         } else {
             throw new ParseException(
                     String.format(MESSAGE_INVALID_COMMAND_FORMAT, CustomiseCommand.MESSAGE_USAGE));
         }
     }
 
     /**
-     * Converts the argument inputs into an array of Attribute.
+     * Converts the input into an array of Attribute objects in the required order.
      *
      * @param args an array of argument inputs
      * @return an array of Attribute
      * @throws ParseException if the user input does not conform the expected format
      */
     private Attribute[] toAttributeOrder(String[] args) throws ParseException {
-        //0 for ADDRESS, 1 for EMAIL, 2 for PHONE, 3 for TAGS
-        boolean[] isAttributeUsed = new boolean[4];
+        boolean[] isAttributeUsed = new boolean[4]; //0: ADDRESS, 1: EMAIL, 2: PHONE, 3: TAGS
+        int noOfMissing = 4 - (args.length - 1); //first element of args is the subCommand
         Attribute[] attributeArr = new Attribute[4];
-        int noOfMissing = 4 - (args.length - 1);
 
-        //First element is ignored as it is the subCommand
+        //first element is ignored as it is the subCommand
         for (int i = 1; i < args.length; i++) {
```

Comment:
This for loop code doesn't seem to be intuitive to understand. Can you add comments here to explain?

Reply:
Ok, I shall add comments and add another index counter `indexOfAttributeArr` to make it more intuitive.